pkg/providers/vsphere/vmprovider_vm_test.go

```diff
@@ -3210,8 +3210,103 @@ func vmTests() {
 						Expect(moVM.Snapshot.CurrentSnapshot).ToNot(BeNil())
 
 						// Find the snapshot name in the tree to verify it matches
-						currentSnapName := vmlifecycle.FindSnapshotNameInTree(moVM.Snapshot.RootSnapshotList, moVM.Snapshot.CurrentSnapshot.Value)
-						Expect(currentSnapName).To(Equal(vmSnapshot.Name))
+						currentSnap := vmlifecycle.FindSnapshotInTree(moVM.Snapshot.RootSnapshotList, moVM.Snapshot.CurrentSnapshot.Value)
+						Expect(currentSnap).ToNot(BeNil())
+						Expect(currentSnap.Name).To(Equal(vmSnapshot.Name))
+					})
+				})
+
+				// Simulate an Imported Snapshot scenario by
+				//	- creating the VC VM and VM while ensuring the backup is not taken.
+				//	- creating a snapshot on VC AND THEN only creating the VMSnapshot CR so that the ExtraConfig is
+				// 	  not stamped by the controller.
+				// 	- change some bits in the VM CR and take a second snapshot. This snapshot can be taken through a
+				//	  VMSnapshot. This is needed to make sure that we are not reverting to a snapshot that the VM is
+				//    running off at the same time.
+				//	- Now, revert the VM to the first snapshot. It is expected that the spec fields would now be approximated.
+				Context("when reverting to imported snapshot", func() {
+					var secondSnapshot *vmopv1.VirtualMachineSnapshot
+
+					BeforeEach(func() {
+						pkgcfg.SetContext(parentCtx, func(config *pkgcfg.Config) {
+							config.Features.VMImportNewNet = true
+						})
+					})
+
+					It("should successfully revert to desired snapshot and approximate the VM Spec", func() {
+						if vm.Labels == nil {
+							vm.Labels = make(map[string]string)
+						}
+
+						vm.Labels[kubeutil.CAPVClusterRoleLabelKey] = ""
+
+						// Create VM first
+						vcVM, err := createOrUpdateAndGetVcVM(ctx, vmProvider, vm)
+						Expect(err).ToNot(HaveOccurred())
+
+						// make sure the VM doesn't have the ExtraConfig stamped
+						var moVM mo.VirtualMachine
+						Expect(vcVM.Properties(ctx, vcVM.Reference(), nil, &moVM)).To(Succeed())
+						Expect(moVM.Config.ExtraConfig).ToNot(BeNil())
+						ecMap := pkgutil.OptionValues(moVM.Config.ExtraConfig).StringMap()
+						Expect(ecMap).ToNot(HaveKey(backupapi.VMResourceYAMLExtraConfigKey))
+
+						// Create first snapshot in vCenter
+						task, err := vcVM.CreateSnapshot(ctx, vmSnapshot.Name, "first snapshot", false, false)
+						Expect(err).ToNot(HaveOccurred())
+						Expect(task.Wait(ctx)).To(Succeed())
+
+						// Create first snapshot CR
+						// Mark the snapshot as ready so that the snapshot workflow doesn't try to create it.
+						conditions.MarkTrue(vmSnapshot, vmopv1.VirtualMachineSnapshotReadyCondition)
```

Comment:
Are you trying to use `CAPVClusterRoleLabelKey ` to skip reconciling snapshot?

Here it doesn't work probably because when you create, it doesn't update the status, you need to call the status API separately to update resource's condition

e.g.:
```
cur := &vmopv1.VirtualMachineSnapshot{}
Expect(ctx.Client.Get(ctx, client.ObjectKeyFromObject(vmSnapshot), cur)).To(Succeed())

conditions.MarkTrue(cur, vmopv1.VirtualMachineSnapshotReadyCondition)
Expect(ctx.Client.Status().Update(ctx, cur)).To(Succeed())
```

Reply:
Right, the snapshot reconcillation is also paused when the label is set.